src/components/Upload.vue

```diff
@@ -110,51 +121,60 @@ export default {
       const results = [];
       this.uploading = true;
       this.errorMessage = null;
-
+      await this.preUpload();
       for (let i = 0; i < this.files.length; i += 1) {
         const file = this.files[i];
         if (file.status === 'done') {
           // We are resuming, skip already completed files
           results.push(file.result);
         } else {
           const progress = (event) => {
-            file.progress = Object.assign({}, file.progress, event);
+            Object.assign(file.progress, event);
```

Comment:
Haven't tested, does reactivity still work properly with this method?

Reply:
Yes. Somehow Object.assign will still call the property setter, making it reactive.